nucliadb/nucliadb/search/requesters/utils.py

```diff
@@ -186,6 +186,28 @@ async def node_query(
 
     error = validate_node_query_results(results or [])
     if error is not None:
+        if (
+            error.status_code >= 500
+            and use_read_replica_nodes
+            and any([node.is_read_replica() for node, _ in queried_nodes])
+        ):
+            # We had an error querying a secondary node, instead of raising an
+            # error directly, retry query to primaries and hope it works
+            logger.warning(
```

Comment:
I wonder if we should capture this and send the error to sentry. Otherwise we will not notice if this ever happens